### ISSUE TITLE
Use only the upper triangle of correlation matrices in summaries/plots

### ISSUE THREAD

**Issue by mike-lawrence** [stan-dev/rstan#329]
Currently, if a model involves a parameter that is a correlation matrix, the various print/summary and plot functions return values/plots for all entries in the matrix, when the diagonal is useless and the lower triangle is redundant. Would be better to keep the index labels (ex. "Omega[1,2]") but omit the entries from the diagonal and lower triangle.

**Comment by bob-carpenter** [comment]
I agree that'd be better output, and we should really be doing that at the Stan level, not the RStan level.  But Ben will probably sort it out in RStan long before we can come up with a general solution in Stan.  We're still waiting on the services refactor before we'd even think about doing this.

**Comment by bob-carpenter** [comment]
What do you think `extract()` should do in that case?  Should that still fill in a full matrix?

**Comment by mike-lawrence** [comment]
Probably best to leave the results of extract() with the full matrix so
that folks can use it for generating new data via MASS::mvrnorm() and the
like.

## 

Mike Lawrence
Graduate Student
Department of Psychology & Neuroscience
Dalhousie University

~ Certainty is (possibly) folly ~

**Comment by bgoodri** [comment]
pairs() drops redundant stuff, but there is no easy way to do it from R. It needs to be done from the Stan side. This is another reason why a symmetric matrix should be a base type.
